File: type-script-nest-graph-ql-실습/src/app/team/resolver/team.resolver.ts
import { Args, Int, Mutation, Query, Resolver } from "@nestjs/graphql";
import { TeamEntity } from "../model/entities/team.entity";
import { Body, Inject, UsePipes, ValidationPipe } from "@nestjs/common";
import { TeamService } from "../service/team-service.interface";
import { DefaultResponse } from "../../common/constant/default.response";
import { TeamResponseDto } from "../model/dto/response/team-response.dto";
import { TeamRequestDto } from "../model/dto/request/team-request.dto";
import { TeamUpdateRequestDto } from "../model/dto/request/team-update.request.dto";
import { TeamAndMemberResponseDto } from "../model/dto/response/team-and-member-response.dto";

@Resolver(() => TeamEntity)
export class TeamResolver {
  constructor(@Inject("TeamService") private readonly teamService: TeamService) {}

  @UsePipes(new ValidationPipe({ transform: true }))
  @Mutation(() => DefaultResponse<string>)
  async saveTeam(@Args("input", { type: () => TeamRequestDto }) @Body() teamRequestDto: TeamRequestDto): Promise<DefaultResponse<number>> {
    return this.teamService.saveTeam(teamRequestDto);
  }

  @Query(() => DefaultResponse<Promise<[TeamEntity[], number]>>)
  async getTeamList(
    @Args("manager", { type: () => String, nullable: true }) manager: string,
    @Args("office", { type: () => String, nullable: true }) office: string,
    @Args("extensionNumber", { type: () => Int, nullable: true }) extensionNumber: string,
    @Args("mascot", { type: () => Int, nullable: true }) mascot: string,
    @Args("cleaningDuty", { type: () => Int, nullable: true }) cleaningDuty: string,
    @Args("project", { type: () => Int, nullable: true }) project: string,
    @Args("pageNumber", { type: () => Int, nullable: true }) pageNumber: number,
    @Args("perPageSize", { type: () => Int, nullable: true }) perPageSize: number,
    @Args("orderBy", { type: () => Boolean, nullable: true }) orderBy: boolean,
  ): Promise<DefaultResponse<TeamResponseDto>> {
    return this.teamService.getTeamList(manager, office, extensionNumber, mascot, cleaningDuty, project, pageNumber, perPageSize, orderBy);
  }

  @Query(() => DefaultResponse<TeamResponseDto>)
  async getTeam(@Args("teamId", { type: () => String }) teamId: number): Promise<DefaultResponse<TeamResponseDto>> {
    return this.teamService.getTeam(teamId);
  }

  @Query(() => DefaultResponse<TeamAndMemberResponseDto>)
  async getTeamByTeamId(@Args("teamId", { type: () => String }) teamId: number): Promise<DefaultResponse<TeamAndMemberResponseDto>> {
    return this.teamService.getTeamByTeamId(teamId);
  }

  @UsePipes(new ValidationPipe({ transform: true }))
  @Mutation(() => DefaultResponse<string>)
  async updateTeam(
    @Args("input", { type: () => TeamUpdateRequestDto }) @Body() teamUpdateRequestDto: TeamUpdateRequestDto,
  ): Promise<DefaultResponse<number>> {
    return this.teamService.updateTeam(teamUpdateRequestDto);
  }

  @Mutation(() => Promise<DefaultResponse<string>>)
  async deleteTeam(@Args("teamId", { type: () => String }) teamId: number): Promise<DefaultResponse<number>> {
    return this.teamService.deleteTeam(teamId);
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/team/resolver/TeamResolver.kt
package com.junyharangstudy.kotlingraphqltest.api.team.resolver

import com.junyharangstudy.kotlingraphqltest.api.common.constant.PageRequestDto
import com.junyharangstudy.kotlingraphqltest.api.team.model.dto.request.TeamCreateRequestDto
import com.junyharangstudy.kotlingraphqltest.api.team.model.dto.request.TeamSearchRequestDto
import com.junyharangstudy.kotlingraphqltest.api.team.model.dto.request.TeamUpdateRequestDto
import com.junyharangstudy.kotlingraphqltest.api.team.model.dto.response.TeamAndMemberResponseDto
import com.junyharangstudy.kotlingraphqltest.api.team.model.dto.response.TeamResponseDto
import com.junyharangstudy.kotlingraphqltest.api.team.service.TeamService
import com.junyharangstudy.kotlingraphqltest.common.constant.DefaultResponse
import org.springframework.graphql.data.method.annotation.Argument
import org.springframework.graphql.data.method.annotation.MutationMapping
import org.springframework.graphql.data.method.annotation.QueryMapping
import org.springframework.stereotype.Controller

@Controller
class TeamResolver(
    private val teamService: TeamService
) {
    @MutationMapping
    fun saveTeam(@Argument("input") teamCreateRequestDto: TeamCreateRequestDto): DefaultResponse<Long> {
        return teamService.saveTeam(teamCreateRequestDto)
    }

    @QueryMapping
    fun getTeamList (
        @Argument teamSearchRequestDto: TeamSearchRequestDto?,
        @Argument pageRequestDto: PageRequestDto?
    ): DefaultResponse<List<TeamResponseDto>> {
        return teamService.getTeamList(teamSearchRequestDto, pageRequestDto)
    }

    @QueryMapping
    fun getTeamAndMemberByTeamId (@Argument teamId: Long): DefaultResponse<TeamAndMemberResponseDto> {
        return teamService.getTeamAndMemberByTeamId(teamId)
    }

    @MutationMapping
    fun updateTeam(
        @Argument("teamId") teamId: Long,
        @Argument("input") teamUpdateRequestDto: TeamUpdateRequestDto): DefaultResponse<Long> {
        return teamService.updateTeam(teamId, teamUpdateRequestDto)
    }

    @MutationMapping
    fun deleteTeam(@Argument teamId: Long): DefaultResponse<Long> {
        return teamService.deleteTeam(teamId)
    }
}